tests/forms_tests/tests/test_forms.py

```diff
@@ -1486,6 +1486,19 @@ class SampleForm(Form):
         f = SampleForm(data={'name': 'bar'})
         self.assertIsInstance(force_text(f['name']), SafeData)
 
+    def test_custom_boundfield(self):
+        bind_to_form_mock = mock.Mock(side_effect=lambda *args, **kwargs: 'custom bound field')
```

Comment:
My feeling is that mock should used as a last resort where other ways of testing don't work. Any thought about this? For example, if I'm looking at the tests to see how to use this feature, it won't really be clear with the mocked version.